blueprints/addon/index.js

```diff
@@ -49,6 +49,9 @@ module.exports = {
     contents.dependencies = contents.dependencies || {};
     contents.devDependencies = contents.devDependencies || {};
 
+		// addon should depend on the ember-source that matches ember-cli
+    contents.dependencies['ember-source'] = contents.devDependencies['ember-cli'];
```

Comment:
```suggestion
    contents.peerDependencies['ember-source'] = '^3.24.0 || ^4.0.0';
```

Another option, maximum compat, matching ember try's config